src/control/cad.js

```diff
@@ -253,103 +261,265 @@ class CadControl extends Control {
+      );
+      return pixelCurrent[0] >= pixelFinal[0] ? coord : finalMax;
+    });
+    const yMin = coordinates.reduce((finalMin, coord) => {
+      const pixelCurrent = this.map.getPixelFromCoordinate(coord);
+      const pixelFinal = this.map.getPixelFromCoordinate(
+        finalMin || coordinates[0],
+      );
+      return pixelCurrent[1] <= pixelFinal[1] ? coord : finalMin;
+    });
+    const yMax = coordinates.reduce((finalMax, coord) => {
+      const pixelCurrent = this.map.getPixelFromCoordinate(coord);
+      const pixelFinal = this.map.getPixelFromCoordinate(
+        finalMax || coordinates[0],
+      );
+      return pixelCurrent[1] >= pixelFinal[1] ? coord : finalMax;
+    });
+
+    // Create four infinite lines through the extremes X and Y and rotate them
+    const minVertLine = new LineString([
+      [xMin[0], -20037508.342789],
+      [xMin[0], 20037508.342789],
+    ]);
+    minVertLine.rotate(this.map.getView().getRotation(), xMin);
+    const maxVertLine = new LineString([
+      [xMax[0], -20037508.342789],
+      [xMax[0], 20037508.342789],
+    ]);
+    maxVertLine.rotate(this.map.getView().getRotation(), xMax);
+    const minHoriLine = new LineString([
+      [-20037508.342789, yMin[1]],
+      [20037508.342789, yMin[1]],
+    ]);
+    minHoriLine.rotate(this.map.getView().getRotation(), yMin);
+    const maxHoriLine = new LineString([
+      [-20037508.342789, yMax[1]],
+      [20037508.342789, yMax[1]],
+    ]);
+    maxHoriLine.rotate(this.map.getView().getRotation(), yMax);
+
+    // Use intersection points of the four lines to get the extent
+    const intersectTopLeft = OverlayOp.intersection(
+      parser.read(minVertLine),
+      parser.read(minHoriLine),
+    );
+    const intersectBottomLeft = OverlayOp.intersection(
+      parser.read(minVertLine),
+      parser.read(maxHoriLine),
+    );
+    const intersectTopRight = OverlayOp.intersection(
+      parser.read(maxVertLine),
+      parser.read(minHoriLine),
+    );
+    const intersectBottomRight = OverlayOp.intersection(
+      parser.read(maxVertLine),
+      parser.read(maxHoriLine),
+    );
+
+    return [
+      [intersectTopLeft.getCoordinate().x, intersectTopLeft.getCoordinate().y],
+      [
+        intersectBottomLeft.getCoordinate().x,
+        intersectBottomLeft.getCoordinate().y,
+      ],
+      [
+        intersectTopRight.getCoordinate().x,
+        intersectTopRight.getCoordinate().y,
+      ],
+      [
+        intersectBottomRight.getCoordinate().x,
+        intersectBottomRight.getCoordinate().y,
+      ],
+    ];
+  }
+
   /**
    * Draws snap lines by building the extent for
    * a pair of features.
    * @private
    * @param {Array.<ol.Feature>} features List of features.
    * @param {ol.Coordinate} coordinate Mouse pointer coordinate.
+   * @param {Array.<number>} pixel Mouse pointer pixel array.
    */
-  drawSnapLines(features, coordinate) {
+  drawSnapLines(features, coordinate, pixel) {
     let auxCoords = [];
```

Comment:
Can you describe what is this array in a comment?

Reply:
done